python/pyarrow/tensor.pxi

```diff
@@ -199,7 +202,13 @@ shape: {0.shape}""".format(self)
             for x in dim_names:
                 c_dim_names.push_back(tobytes(x))
 
-        coords = np.vstack([obj.row, obj.col]).T
+        row = obj.row
+        col = obj.col
+        if obj.has_canonical_format:
+            order = np.lexsort((col, row))  # row-major order
```

Comment:
Sorting indices is necessary to conserve the canonical format.

Reply:
I don't understand this. If Scipy says the indices are sorted, why do you need to sort them again?

Reply:
SciPy's coo_matrix employs column-major order, but our SparseCOOIndex employs row-major order.
This is the reason why we need to sort the index here.

The reason why our SparseCOOIndex should employ row-major order is that TensorFlow's SparseTensor employs row-major order.  Converting from SciPy's coo_matrrix needs to copy the index matrix even if we employ column-major order, but converting from TensorFlow's SparseTensor does not need to copy the index matrix if we employ row-major order.

Reply:
So column-major is scipy-canonical and row-major is tensorflow canonical?
Would we then not rather just introduce a new property e.g.: `has_duplicates`? I would find that less complex.

Reply:
The idea of canonical COO index IMO unnecessarily hides the two properties (presence of duplication and row-ordered-ness).